### PR TITLE
Add date search, restyle

### PR DESCRIPTION
**Pull request by jpalmieri**
Review this after https://github.com/jpalmieri/macro_search/pull/1 is merged.

Now users can search by created and updated date as well. Also made it look nicer using bootstrap. This has some refactor potential, but I'll take care of that later.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -63,6 +71,18 @@
...
       }
+      if ( this.$('.check.created').is(':checked') ) {
+        results = this.filterByCreatedDate(results);
+      }
+      if ( this.$('.check.updated').is(':checked') ) {
+        results = this.filterByUpdatedDate(results);
+      }
+
+      // Remove times from dates
+      _.each(results, function(macro) {
```
Comment (PolRod) [inline app.js:82]: Why did you change all the forEach methods with the underscore equivalent in this PR?

File: app.js [hunk 2]
```diff
@@ -90,29 +112,59 @@
...
-      macros.forEach( function(macro) {
+      _.each(macros, function(macro) {
         var comments = this.getComments(macro);
-        comments.forEach( function(comment) {
-          if ( comment.indexOf(query) > -1 ) {
-            results.push(macro);
-          }
-        });
+        // Filter our comments which don't match query
+        comments = _.filter(comments, function(comment) { return comment.indexOf(query) > -1; });
```
Comment (PolRod) [inline app.js:120]: noiiice

**Comment by PolRod** [comment]
:fire::mag:

File: app.js (hunk 1 above)
Comment (jpalmieri) [inline app.js:82]: I just snuck it into this PR, it doesn't really change anything except style.

I changed it for continuity (since I use underscore methods elsewhere now) and for style, since ZD apps best practices told me to: https://github.com/zendesk/demo_apps/blob/master/Bestpractices.md

I personally like the `.forEach` function better since it reads more like ruby, but I guess this isn't ruby :crying_cat_face:

**Comment by jpalmieri** [comment]
:arrow_up: YES
